src/LiveComponent/src/LiveComponentHydrator.php

```diff
@@ -663,7 +663,8 @@ private function processOnUpdatedHook(object $component, string $frontendName, L
 
         foreach ($onUpdated as $propName => $funcName) {
             if (LiveProp::IDENTITY === $propName) {
-                if (!$dehydratedUpdatedProps->hasPropValue($frontendName)) {
+                if (!$dehydratedUpdatedProps->hasPropValue($frontendName)
+                    && !$dehydratedUpdatedProps->hasNestedPathsForProperty($frontendName)) {
                     continue;
```

Comment:
I'm not sure this is the right place... for a LiveProp IDENTITY the only thing that can change is... the prop itself no ? Is this not something we should do line 680+ ?

Reply:
I don't think thats the best place either to check if there are changes changes to a array property. My intention was to tigger the onUpdated whenever there is a change within the array without providing a specific nested path.

Reply:
Maybe I'm wrong, but it's also strange that if the `$propMetadata->onUpdated()` is a string, the update hook only triggers if the property has been completely changed to another property and not if part of it has been modified.

Reply:
I get that, but i fear doing so here will have side effects .. wdyt ?